rspace/src/main/scala/coop/rchain/rspace/nextgenrspace/RSpace.scala

```diff
@@ -34,9 +33,16 @@ class RSpace[F[_], C, P, A, R, K] private[rspace] (
     contextShift: ContextShift[F],
     scheduler: ExecutionContext,
     metricsF: Metrics[F]
-) extends RSpaceOps[F, C, P, A, R, K](store, branch)
+) extends RSpaceOps[F, C, P, A, R, K](storeAtom, branch)
     with ISpace[F, C, P, A, R, K] {
 
+  //TODO close in some F state abstraction
+  val historyRepositoryAtom: AtomicAny[HistoryRepository[F, C, P, A, K]] = AtomicAny(
+    historyRepository
+  )
+
+  def store: HotStore[F, C, P, A, K] = storeAtom.get()
```

Comment:
Consider replacing the parameter names order:
1. C with A
2. P with C
3. K with P
That should do it